### DISCUSSION TITLE
Calling `#update` on a dataset doesn't update the `updated_at` column

### DISCUSSION THREAD

**Discussion by AlasdairWallaceMackie** [jeremyevans/sequel#2171] [General]
As the title says, this is something I noticed today. Wanted to know if this was intended behavior before I opened an issue. Right now my workaround is is to use `#map` on a dataset to individually update each row, in order to get their `updated_at` columns updated as expected.

Snippets:
```
# Update dataset
ruby-dev >> User.where(id: ids).update(is_admin: false)
=> 5

# Check values for `updated_at` - no change
ruby-dev >> User.where(id: ids).map(&:refresh).map(&:updated_at)
=> [2023-01-13 14:24:46.571127 UTC, 2023-09-28 21:38:28.442335 UTC, 2024-03-19 19:40:03.279699 UTC, 2024-05-14 19:24:06.390158 UTC, 2024-03-19 19:40:03.235151 UTC]
```

```
# Update dataset by mapping through each row
ruby-dev >> User.where(id: ids).map { |user| user.update(is_admin: true) }

# Check values for `updated_at` - All have been updated
ruby-dev >> User.where(id: ids).map(&:refresh).map(&:updated_at)
=> [2024-05-29 17:30:12.076985 UTC, 2024-05-29 17:30:14.376477 UTC, 2024-05-29 17:30:13.464115 UTC, 2024-05-29 17:30:15.172423 UTC, 2024-05-29 17:30:16.104969 UTC]
```

**Comment by adam12** [comment]
I'm assuming you're using the `timestamps` model plugin? If so, I'm almost certain this is intended behaviour.

An alternative, if you're using Postgres, is `sequel_postgres_triggers` plugin from Jeremy. You could do this with your own triggers as well if you're on another database.

**Comment by jeremyevans** [comment]
As @adam12 wrote, this is expected behavior.  `Dataset#update` is a dataset-level update (no model save hooks), `Model#update` is a model-level update (with model save hooks).

**Comment by AlasdairWallaceMackie** [comment]
Thanks for the quick reply! I will close this discussion
